src/corelib/tools/qarraydata.h

```diff
@@ -96,7 +96,7 @@ struct Q_CORE_EXPORT QArrayData
         // Q_ASSERT(offset > 0 && offset < 100 && "Globals should use a pointer!");
         // Seems like the generated QMetaObject tables use larger offsets so just assert that the length is sensible
         Q_ASSERT(offset < __builtin_cheri_length_get(obj));
-        ret = __builtin_cheri_offset_increment(obj, qint64(offset));
+        ret = const_cast<void *>(__builtin_cheri_offset_increment(obj, qint64(offset)));
```

Comment:
Ah yeah, fallout from adding type preservation to the builtins. But `__builtin_cheri_offset_increment` is a bit pointless (I go back and forth on whether we should even provide it; if not it'd be the only instruction without an intrinsic, but it also shouldn't be used). Could you instead rewrite this to just do pointer arithmetic on a `const char *`?

Reply:
Yeah I don't recall why I used the intrinsic originally. Fortunately most of this diff is gone in the Qt6 version.

Reply:
Rewrote it to do it that way.